src/main/java/com/arpnetworking/metrics/StopWatch.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2017 Inscope Metrics, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics;
+
+import com.arpnetworking.metrics.impl.TsdStopWatch;
+
+/**
+ * Interface for a thread safe stop watch. The stop watch starts on creation
+ * and must be stopped to access the elapsed time. Once stopped the instance
+ * must be discarded and cannot be started again. The design is minimal to
+ * reduce contention at the cost of object creation because timers in this
+ * library represent samples and do not reuse their underlying stop watches.
+ * However, it is possible to build a stateful timer around this but you
+ * will need to add stricter synchronization or else give up thread safety.
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+public interface StopWatch {
+
+    /**
+     * Create and start a new {@link StopWatch} instance.
+     *
+     * @return new started {@link StopWatch} instance.
+     */
+    static StopWatch start() {
```

Comment:
So our implementation classes are generally hidden, that is they don't have public constructors. Most of the time the instances are created by the Metrics instance which acts as a sort of factory (e.g. for TsdTimer and TsdCounter). Many other classes are not created at all (e.g. Units). This breaks down when we want to enable clients to create these classes. Here the interface acts as a factory by instantiating a default StopWatch. Making the constructor public on TsdStopWatch exposes the class, so the alternative would be an explicit factory class which is overkill. As much as it has turned my stomach as well, it has become a rather common pattern accepted by many people/groups we look up to. For example, it's in the Guava StopWatch ;)